### ISSUE TITLE
[withMenuContext] Unnecessary re-renders?

### ISSUE THREAD

**Issue by joncursi** [instea/react-native-popup-menu#103]
I am using `withMenuContext` to provide the `ctx` prop on my screen components. However, I've been noticing sluggish behavior on all of my screens that use`withMenuContext`:

```jsx
import { withMenuContext } from 'react-native-popup-menu';

class SomeComponent extends React.Component {
  render() {
    console.log('re-render');
    return (...);
  }
}

const enhance = compose(
  withMenuContext,
);

export default enhance(SomeComponent);
```

After doing some `console.log`ging, I noticed that a lot of additional re-renders are happening when `withMenuContext` is being used in the HOC hierarchy. So much so, that my screens that use large `FlatList`s run into memory issues:

```
Maximum update depth exceeded
```

I'm on version `0.13.1` and this seems to be a new issue with the new `ctx` workflow for interacting with menus. I didn't run into this problem on the `0.12.x` releases.

**Comment by sodik82** [comment]
Thanks for reporting.

We have switched to new React context API, however the context value is basically singleton (does not change over time and we are not rerendering provider neither), thus I was not expecting any troubles here.

Would you mind maybe checking with help of `Menu.debug=true` what is going on more deeply? 

Tips for workarounds: 
1. Either pull down the context HOC down in your component hierarchy to reduce re-renders.
2. Try to mnemoize context value (as it does not change over time)

**Comment by sodik82** [comment]
@joncursi can you tell us something more under which circumstances are the re-render happening? 

In our simplified example I got only one render and also checking the React documentation indicates, that context consumer should "render" only once (unless re-rendered from the parent) as the context value is singleton

**Comment by sodik82** [comment]
closing for inactivity. If you find out more about the issue, will reopen.
